Tests/JobsTests/MetricsTests.swift

```diff
@@ -0,0 +1,294 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Hummingbird server framework project
+//
+// Copyright (c) 2021-2024 the Hummingbird authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See hummingbird/CONTRIBUTORS.txt for the list of Hummingbird authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Atomics
+import Jobs
+import Logging
+import Metrics
+import NIOConcurrencyHelpers
+import ServiceLifecycle
+import XCTest
+
+final class TestMetrics: MetricsFactory {
```

Comment:
I've just made some changes to this to make it compile in Swift 6 mode. You'll need to wrap all this mutable state in NIOLockedValueBoxes. If you just copied and pasted this from Hummingbird, you can probably do the same again. Although you will need to edit the accesses in the `XCTAssert...` calls

Reply:
> I've just made some changes to this to make it compile in Swift 6 mode. You'll need to wrap all this mutable state in NIOLockedValueBoxes. If you just copied and pasted this from Hummingbird, you can probably do the same again. Although you will need to edit the accesses in the `XCTAssert...` calls

Thanks Adam! I have updated the helpers with the help of your previous PR.